aten/src/TH/THTensor.cpp

```diff
@@ -54,3 +56,69 @@ void THTensor_free(THTensor *self)
     }
   }
 }
+
+// On a high level,
+// 1. separate tensor->size into chunks of dimensions, where the dimensions are
+//    ``contiguous'' in each chunk, i.e., stride[i] = size[i+1] * stride[i+1]
+// 2. view_size must be able to be separated into same number of chunks as tensor->size was separated into,
+//    where each chunk of view_size has matching ``numel'', i.e., number of subspaces,
+//    as the corresponding chunk of tensor->size.
+at::optional<std::vector<int64_t>>
```

Comment:
What does the nullopt case signify?

Reply:
I guess, not possible to restride for the new shape.

Reply:
Right, I just changed it from `{}` to `nullopt` so it's more explicit -- `{}` is a valid initializer for both `optional` and `vector`.